server/src/test/java/org/opensearch/cluster/service/ThrottlerTests.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.cluster.service;
+
+import org.opensearch.common.AdjustableSemaphore;
+import org.opensearch.test.OpenSearchTestCase;
+
+/**
+ * Contains tests of {@link Throttler}
+ */
+public class ThrottlerTests extends OpenSearchTestCase {
+
+    public void testDisabledThrottling() {
+        Throttler throttler = new Throttler(true);
+        throttler.updateThrottlingLimit("testKey", 1);
+        throttler.setThrottlingEnabled(false);
+        boolean firstCall = throttler.acquire("testKey", 5);
+        assertTrue(firstCall);
+    }
+
+    public void testThrottling() {
+        Throttler throttler = new Throttler(true);
+        throttler.updateThrottlingLimit("testKey", 5);
+        boolean firstCall = throttler.acquire("testKey", 1);
+        assertTrue(firstCall);
+
+        boolean secondCall = throttler.acquire("testKey", 2);
+        assertTrue(secondCall);
+
+        boolean thirdCall = throttler.acquire("testKey", 4);
+        assertFalse(thirdCall);
+
+        // will remove used value
+        throttler.release("testKey", 2);
+        boolean fourthCall = throttler.acquire("testKey", 4);
```

Comment:
with every call, please add in the comment how are permits are acquired now, would be easy to follow test with that.

Reply:
Sure will add it.